docs/docs/components/migrating/README.md

```diff
@@ -0,0 +1,190 @@
+
+**Simplified discovery**
+Actions you publish are now grouped under **My Actions** when adding a step to a workflow. NOTE: this option will appear in the workflow builder *after* you publish your first action.
+
+## Getting Started
+
+Ready to develop your first component action? We recommend starting with our [quickstart guide](/components/quickstart/nodejs/actions/). Then review both our [component API reference](/components/api) and [actions published to Pipedream’s Github repo](https://github.com/pipedreamhq/pipedream/compnents).
+
+## Migration Example
+
+Let's walk through an example that migrates code for a legacy action to a Pipedream component. 
+
+### Legacy Code Example
+
+Following is the code for the legacy action to get a Github repo (Github was linked to this action via Pipedream's UI, so it's not declared in the code):
+
+```javascript
+const config = {
+  url: `https://api.github.com/repos/${params.owner}/${params.repo}`,
+  headers: {
+    Authorization: `Bearer ${auths.github.oauth_access_token}`,
+  }
+}
+return await require("@pipedreamhq/platform").axios(this, config)
+```
+
+Also, following is the associated JSON schema that defines metadata for the `params` inputs:
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "owner": {
+      "type": "string",
+      "description": "Name of repository owner."
+    },
+    "repo": {
+      "type": "string",
+      "description": "Name of repository."
+    }
+  },
+  "required": [
+    "owner",
+    "repo"
+  ]
+}
+```
+
+### Converting to the Component Model
+
+To convert the code above to the component model, we need to:
+
+1. Link the Github app to the component using `props` (so we can use Pipedream managed auth for Github)
+2. Define `props` for `owner` and `repo` so we can capture user input. The definition for each prop includes the `type` and `description` metadata. Additionally, since all the fields are required, we do not need to set the `optional`  property (set `optional` to `true` for optional `props`). This metadata was previously captured in the JSON schema.
+3. Replace references to `params` in the `run()` method. `props` are bound to `this`. 
+4. Update the reference to the Github OAuth token from `auths.github.oauth_access_token` to `this.github.$auth.oauth_access_token` (note: `github` in this context references the name of the prop, not the name of the app; if the prop was named `gh` then the auth would be referenced via `this.gh.$auth.oauth_access_token`).
+5. Replace the `@pipedreamhq/platform`  npm package with the standard `axios` package
+
+```javascript
+const axios = require("axios")
+
+module.exports = {
+  type: "action",
+  name: "Get Repo Example",
+  key: "github_get-repo-example",
+  version: "0.0.1",
+  props: {
```

Comment:
Might be worth cleaning up the indentation in this code.

Reply:
that's really strange. the indentation looks correct in my editor. I'll check the preview.

Reply:
i can see the formatting issues in the preview. i'll try fixing manually -- there may be a copy/paste artifact I can't see in my editor.